sn_networking/src/bootstrap.rs

```diff
@@ -49,16 +49,24 @@ impl SwarmDriver {
     }
 
     pub(crate) fn trigger_network_discovery(&mut self) {
+        let now = Instant::now();
         // The query is just to trigger the network discovery,
         // hence no need to wait for a result.
-        for addr in &self.network_discovery_candidates {
-            let _ = self
+        for addr in self.network_discovery.candidates() {
+            let query_id = self
                 .swarm
                 .behaviour_mut()
                 .kademlia
                 .get_closest_peers(addr.as_bytes());
+            let _ = self.pending_get_closest_peers.insert(
+                query_id,
+                (PendingGetClosestType::NetworkDiscovery, Default::default()),
+            );
         }
+        // Refresh the candidate list to not query the same candidates over and over again.
+        self.network_discovery.try_refresh_candidates();
```

Comment:
maybe call something like `for addr in self.network_discovery.candidates()` above,
which internally refresh the candidate by itself, will be better ?
i.e. avoid calling seperate `refresh`